doboto/Endpoint.py

```diff
@@ -80,3 +83,65 @@ def pages(self, request_url, expect, params=None):
                 next_url = None
 
         return items
+
+    def action_result(self, action, wait, poll, timeout):
+        """
+        General action result processor for waiting
+        """
+
+        if not wait:
+            return action
+
+        if poll < 1:
+            poll = 1
+
+        start_time = time.time()
+
+        while action["status"] == "in-progress":
+
+            time.sleep(poll)
+            try:
+                action = self.do.action.info(action["id"])
+            except Exception as exception:
+                if time.time() - start_time > timeout:
+                    raise DOBOTOPollingException(polling=action, error=exception)
```

Comment:
I think we want an `else` clause here, don't we?  Something along the lines of:

```
else:
    raise DOBOTOException(polling=action, error=exception)
```

I thought that's what we had agreed on, simply because this is still going to hide any unexpected exceptions.